snapcraft/tests/test_commands_clean.py

```diff
@@ -14,168 +14,122 @@
-
-        patcher = mock.patch('os.rmdir')
-        self.mock_rmdir = patcher.start()
-        self.addCleanup(patcher.stop)
-
-        self.clean_calls = []
-
-        class FakePart:
-
-            def __init__(self, name, partdir):
-                self.name = name
-                self.partdir = partdir
-
-            def names(self):
-                return [self.name, ]
-
-            def clean(s):
-                self.clean_calls.append(s.name)
-
-        class FakeConfig:
-            all_parts = [
-                FakePart('part1', 'partdir1'),
-                FakePart('part2', 'partdir2'),
-                FakePart('part3', 'partdir3'),
-            ]
-
-        self.fake_config = FakeConfig()
-
-        patcher = mock.patch('snapcraft.yaml.load_config')
-        self.mock_load_config = patcher.start()
-        self.mock_load_config.return_value = self.fake_config
-        self.addCleanup(patcher.stop)
+    yaml_template = """name: clean-test
+version: 1.0
+vendor: To Be Removed <vendor@example.com>
+summary: test clean
+description: if the clean is succesful the state file will be updated
+icon: icon.png
+
+parts:
+{parts}"""
+
+    yaml_part = """  clean{:d}:
+    plugin: nil
+    source: ."""
+
+    def make_snapcraft_yaml(self, n=1, create=True):
+        parts = '\n'.join([self.yaml_part.format(i) for i in range(n)])
+        super().make_snapcraft_yaml(self.yaml_template.format(parts=parts))
+        open('icon.png', 'w').close()
+
+        parts = []
+        for i in range(n):
+            part_dir = os.path.join(common.get_partsdir(), 'clean{}'.format(i))
+            state_file = os.path.join(part_dir, 'state')
+            parts.append({
+                'part_dir': part_dir,
+                'state_file': state_file,
+            })
+            if create:
+                os.makedirs(part_dir)
+                open(state_file, 'w').close()
+
+        if create:
+            os.makedirs(common.get_stagedir())
+            os.makedirs(common.get_snapdir())
+
+        return parts
 
     def test_clean_all(self):
-        clean.main()
+        parts = self.make_snapcraft_yaml(n=3)
 
-        self.mock_exists.assert_has_calls([
-            mock.call(common.get_partsdir()),
-            mock.call().__bool__(),
-            mock.call(common.get_stagedir()),
-            mock.call().__bool__(),
-            mock.call(common.get_snapdir()),
-            mock.call().__bool__(),
-        ])
+        clean.main()
 
-        self.mock_rmtree.assert_has_calls([
-            mock.call(common.get_stagedir()),
-            mock.call(common.get_snapdir()),
-        ])
+        for part in parts:
+            self.assertFalse(
+                os.path.exists(part['part_dir']),
+                'Expected for {!r} to be wiped'.format(part['part_dir']))
+            self.assertFalse(
+                os.path.exists(part['state_file']),
+                'Expected for {!r} to be wiped'.format(part['state_file']))
```

Comment:
This might be unnecessary, isn't it enough to check that partsdir doesn't exist?
Feel free to leave the extra checks if you think they might be useful.